src/citationnet/static/force-citationNet.js

```diff
@@ -0,0 +1,419 @@
+* @param {String} path - path or url to JSON file
+* @returns {object} parsed JSON object
+*/
+async function fetchJSON(data) {
+    // console.log('parsed', data)
+
+    try {
+        // waits until the request completes...
+        const retdata = await JSON.parse(data);
+    } catch (error) {
+        console.log(data);
+        throw error;
+    }
+    return retdata;
+}
+
+
+class CitationNet {
+
+    /**
+    * Constructs a new CitationNet object, but does not initialize it. Call object.initialize() right after this.
+    * @param {String} jsondata - path or url to citation data as JSON file/stream
+    */
+    constructor(jsondata = null) {
+        // if jsonPath is not provided try to get globJsonPath, show alert if fails
+        try {
+            if (!(jsondata)) jsondata = jsondata;
+        } catch (error) {
+            alert("no JSON containing citation data specified, graph cannot be displayed");
+        }
+
+        this.jsondata = jsondata;
+        this.is_initialized = false;
+    }
+
+    /**
+    * Fetches and processes data, initializes graph and sets view. Constructors cannot be async in JS, which is needed for fetching and saving data.
+    */
+    async initialize(make_cylinder = false) {
+        this.container = document.getElementById('3d-graph');
+        // fetch data (async)
+        // this.data = await fetchJSON(this.jsonPath);
+        // this.processData();
+        await this.getStats();
+        this.makeGraph(this.data);
+
+        // variables for control toggle-functions
+        this.nodeSize = false;
+        this.edgesOnlyInput = true;
+        this.distanceFromInputNode = false;
+
+        this.adaptWindowSize();
+        this.view('side');
+        this.toggleNodeSize();
+
+        if (make_cylinder) {
+            this.makeCylinder();
+            this.graph.controls()._listeners.change.push(this.renderLabels());
+        }
+
+        this.is_initialized = true;
+    }
+
+    /**
+    * Reads current window size, adapts canvas size and camera projection settings  to it.
+    */
+    adaptWindowSize() {
+        this.graph.height(window.innerHeight - document.getElementsByClassName("navbar")[0].scrollHeight);
+        this.graph.width(window.innerWidth);
+
+        this.graph.camera().left = this.graph.width() / -2;
+        this.graph.camera().right = this.graph.width() / 2;
+        this.graph.camera().top = this.graph.height() / 2;
+        this.graph.camera().bottom = this.graph.height() / -2;
+        this.graph.camera().updateProjectionMatrix();
+    }
+
+    /**
+    * Instantiate and configure graph object
+    * @returns {function} Brief description of the returning value here.
+    */
+    makeGraph() {
+        // sort nodes descending by data.nodes.attributes['ref-by-count'], get first items 'ref-by-count'-attribute
+        const maxCites = this.data.nodes.sort((a, b) => (a.attributes['ref-by-count'] > b.attributes['ref-by-count']) ? -1
```

Comment:
This is compact, but very hard to read. Custom sorting can be messy in JS. I would recommend expanding the function with a code block so that there's sufficient room to describe the steps. Just make sure to `return` the value.